src/subcommands/utils.go

```diff
@@ -67,22 +90,96 @@ func writeIndex(indexFilePath string, indexMap map[string]*pb.IndexListEntry) er
 	return nil
 }
 
-func makeCopy(indexPath string, ile *pb.IndexListEntry) error {
+// fetchFromSource retrieves content from either a local file or a webpage based on the provided source type.
+//
+// Parameters:
+//   - sourceType: The type of source (pb.SourceType).
+//   - uri: The location of the content (file path or URL).
+//
+// Returns:
+//   - []byte: The content retrieved from the source.
+func fetchFromSource(sourceType pb.SourceType, uri string) ([]byte, error) {
```

Comment:
You should put this in a new file or even a new directory (alongside subcommands) because there will be a lot of complex logic to handle this across source types